src/app/models/AudioRecording.ts

```diff
@@ -29,40 +29,32 @@ export interface AudioRecordingInterface {
 /**
  * An audio recording model
  */
-export class AudioRecording implements AudioRecordingInterface {
+export class AudioRecording extends AbstractModel
+  implements AudioRecordingInterface {
   public readonly kind: "AudioRecording";
-  public readonly id: Id;
-  public readonly uuid: Uuid;
-  public readonly recordedDate: DateTimeTimezone;
-  public readonly siteId: Id;
-  public readonly durationSeconds: number;
-  public readonly sampleRateHertz: number;
-  public readonly channels: number;
-  public readonly bitRateBps: number;
-  public readonly mediaType: string;
-  public readonly dataLengthBytes: number;
-  public readonly status: "ready" | "uploading" | "corrupt";
-  public readonly createdAt: DateTimeTimezone;
-  public readonly updatedAt: DateTimeTimezone;
+  public readonly id?: Id;
+  public readonly uuid?: Uuid;
+  public readonly recordedDate?: DateTimeTimezone;
+  public readonly siteId?: Id;
+  public readonly durationSeconds?: number;
+  public readonly sampleRateHertz?: number;
+  public readonly channels?: number;
+  public readonly bitRateBps?: number;
+  public readonly mediaType?: string;
+  public readonly dataLengthBytes?: number;
+  public readonly status?: "ready" | "uploading" | "corrupt";
+  public readonly createdAt?: DateTimeTimezone;
+  public readonly updatedAt?: DateTimeTimezone;
```

Comment:
This is a good change. 

I think generally we'll have more property getters and setters defined though.

Hopefully we can use those to convert types (like strings to dates) during serialization